OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -133,21 +130,56 @@ public OpenAlSoundEngine(string deviceName)
 			context = ALC10.alcCreateContext(device, null);
 			if (context == IntPtr.Zero)
 				throw new InvalidOperationException("Can't create OpenAL context");
+
 			ALC10.alcMakeContextCurrent(context);
 
-			for (var i = 0; i < PoolSize; i++)
+			var size = new int[1];
+			ALC10.alcGetIntegerv(device, ALC10.ALC_ATTRIBUTES_SIZE, 1, size);
+			if (CheckError($"Failed to query OpenAL attributes using {deviceName}"))
+				return;
+
+			var values = new int[size[0]];
+			ALC10.alcGetIntegerv(device, ALC10.ALC_ALL_ATTRIBUTES, size[0], values);
+			if (CheckError($"Failed to query OpenAL attributes using {deviceName}"))
+				return;
+
+			for (var i = 0; i < values.Length; i++)
+				if (values[i] == ALC11.ALC_MONO_SOURCES || values[i] == ALC11.ALC_STEREO_SOURCES)
+					if (values[i + 1] > poolSize)
+						poolSize = values[i + 1];
```

Comment:
Shouldn't this be set to the smaller of mono/stereo sources?

Reply:
for me it returns just 1 stereo. Ah I had a bad setup earlier so the current code that I pushed wasn't working, I'm now adding them up as suggested in by the main al soft dev https://github.com/kcat/openal-soft/issues/736#issuecomment-1188400131